### ISSUE TITLE
AzureMonitor: template variables

### ISSUE THREAD

**Issue by zensqlmonitor** [grafana/grafana#33834]
**What happened**:
While other functions (Namespaces and ResourceNames) are properly working, the function MetricNames is not working with a resource group that contains hyphen. 
Example: RG-WEU-001.

I got the following error: 
code: "SEM0100", message: "'' operator: Failed to resolve scalar expression named 'RG'

**How to reproduce it (as minimally and precisely as possible)**:
Define a variable and use the function **MetricNames(aResourceGroup, aNamespace, aResourceName)** with aResourceGroup equal to RG-WEU-001.

**Environment**:
- Grafana version: Grafana Cloud
- Data source type & version: Azure Monitor

**Comment by sunker** [comment]
This is confusing - according to the [docs](https://grafana.com/docs/grafana/latest/datasources/azuremonitor/#create-template-variables-for-metrics), the MetricNames query is supposed to have three or four args, but according to the [code](https://github.com/grafana/grafana/blob/97cf00ab33c6bcec0df87ff9da7d714bf92c2a59/public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_monitor/azure_monitor_datasource.ts#L289-L311) it looks like it's expecting 4 or 5 args, one being metric namespace which it not documented.  

Can you ever get the MetricNames(..args) query function to work? Even when using a  resource group without hyphens?

**Comment by zensqlmonitor** [comment]
Thanks for your reply.
No, indeed. Also not working with a resource group without hyphens...

operator: Failed to resolve table or scalar expression named 'rgMyRG'

**Comment by sarahzinger** [comment]
Hello! I was able to confirm the issue seems to be in the documentation. I was able to get MetricNames to return some data by calling:
```
MetricNames(resourceGroup,metricDefinition,resourceName,metricNamespace)
```
@zensqlmonitor want to give that a try and see if it works for you? In the mean time we have a ticket to refresh documentation for azure monitor and have referenced this issue so that we'll be sure to address it! Thanks for reporting!
